src/main/java/duke/sonautil/Storage.java

```diff
@@ -0,0 +1,226 @@
+package duke.sonautil;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+
+/**
+ * Works on loading, changing and maintaining the contents in the hard disk file
+ */
+public class Storage {
+    private final String path;
+
+    public Storage(String path) {
+        this.path = path;
+    }
+
+    /**
+     * Converts the command into hard disk file for record.
+     *
+     * @param command details of command retrieved from Parser
+     * @throws IOException if writeToFile/changeMarking fails
+     */
+    public void executeCommand(String[] command) throws IOException {
+
+        String keyword = command[0];
+        switch (keyword) {
+        case "todo":
+            writeToFile("T | 0 | " + command[1]);
+            break;
+        case "event":
+            writeToFile("E | 0 | " + command[1] + " | " + command[2]);
+            break;
+        case "deadline":
+            writeToFile("D | 0 | " + command[1] + " | " + command[2]);
+            break;
+        case "delete":
+            deleteFromFile(Integer.parseInt(command[1]));
+            break;
+        case "mark":
+            changeMarking("0", "1", Integer.parseInt(command[1]));
+            break;
+        case "unmark":
+            changeMarking("1", "0", Integer.parseInt(command[1]));
+            break;
+        default:
+        }
+    }
+
+    /**
+     * deletes the Duke.task from local file by creating a temp file
+     * and copy every Duke.task except the one to be deleted
+     *
+     * @param taskToDelete the index of Duke.task to delete
+     */
```

Comment:
The "Duke." before each reference to tasks and other Duke related functions is most likely due to a refactoring bugs (i.e. from changing the name of a package, or moving a file to a new package). I noticed the same issue elsewhere in the code documentation. Consider deleting each of these modifications by the program in the interest of clarity.